### PR TITLE
Retry transaction without for update

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

**Comment by coocood** [comment]
It seems like "ForUpdate" is never reset. If user executed an "FoUpdate" transaction, then the following transactions would be always "ForUpdate".

**Comment by ngaut** [comment]
PTAL

File: session.go [hunk 1]
```diff
@@ -126,8 +165,59 @@ func (s *session) String() string {
...
+	}()
+
+	if forUpdate := s.Value(forupdate.ForUpdateKey); forUpdate != nil {
+		return errors.Errorf("can not retry select for update statement")
+	}
+
+	var err error
+	for {
+		// Clear history
+		s.history.history = s.history.history[:0]
```
Comment (shenli) [inline session.go]: s.history.reset()?

File: driver.go [hunk 2]
```diff
@@ -239,8 +241,13 @@ func (c *driverConn) Commit() error {
 	if c.s == nil {
 		return qerror.ErrCommitNotInTransaction
 	}
+	_, err := c.s.Execute(txCommitSQL)
 
-	if _, err := c.s.Execute(txCommitSQL); err != nil {
+	if errors2.ErrorEqual(err, kv.ErrConditionNotMatch) {
+		return c.s.Retry()
```
Comment (shenli) [inline driver.go]: Change the error type of unique key conflict.

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**